File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool palindrome(string x)
    {
        int low = 0;
        int high = x.size() - 1;


        while(low < high)
        {
            if(x[low] != x[high])
            {
                return false;
            }

            low++;
            high--;
        }

        return true;
    }


    bool isPalindrome(int x) {
        if(x<0)
        {
            return false;
        }
        
        return palindrome(to_string(x));
    }
};

File: 9-palindrome-number/palindrome-number.py
class Solution:

    def palindrome(seld, x: str) -> bool:
        low = 0
        high = len(x) - 1

        while(low<high):
            if(x[low] != x[high]):
                return False
            
            low+=1
            high-=1
        
        return True

    def isPalindrome(self, x: int) -> bool:
        if(x < 0):
            return False
        
        return self.palindrome(str(x))
        